django_mesh/models.py

```diff
@@ -28,37 +29,62 @@
 
 # 3d party imports
 from model_utils import Choices
+import markdown
+import textile # to do: add oembed for textile markup
+from pyembed.markdown import PyEmbedMarkdown
+from pyembed.core import PyEmbed
+from bs4 import BeautifulSoup
 
 # App imports
 from .managers import PostQuerySet, ChannelQuerySet, TagQuerySet
 
-oembed_regex = re.compile(r'^(?P<spacing>\s*)(?P<url>http://.+)', re.MULTILINE)
-
 @python_2_unicode_compatible
 class _Abstract(models.Model):     #microblog compatible.
     slug = models.SlugField(unique=True)
     title = models.CharField(max_length=140, unique=True)
     text = models.TextField(default='')
     rendered_text = models.TextField(default='', blank=True)
+    CONVERSION = Choices(
+        (0, 'PLAINTEXT', 'Plaintext',),
+        (1, 'MARKDOWN', 'Markdown',),
+        (2, 'TEXTILE', 'Textile'),
+        )
+
+    conversion      = models.IntegerField(max_length=1, default=CONVERSION.PLAINTEXT, choices=CONVERSION)
+
+    def render(self, *args, **kwargs):
+
+        if self.conversion == self.CONVERSION.PLAINTEXT:
+
+            self.rendered_text = cgi.escape(self.text, True)
+
+        elif self.conversion == self.CONVERSION.MARKDOWN:
+
+            soup = BeautifulSoup(markdown.markdown(self.text))
+
+            for url in soup.find_all(text=re.compile('http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+')):
+                if url.find_parent('p'):
+                    url.wrap(soup.new_tag('a', href=url))
+
+            for link in soup.find_all('a'):
```

Comment:
I think my original intent was to only use oembed on plain URLs in the text, so the content of this for loop can be moved to the previous for loop. As is, it will also oembed URLs that were explicit made links by the user.

I could be wrong though, I might be remembering wrong.